sync/class.jetpack-sync-client.php

```diff
@@ -798,23 +801,43 @@ function reset_sync_queue() {
 		$this->sync_queue->reset();
 	}
 
+	function get_settings() {
+		$settings = array();
+		foreach( array_keys( self::$valid_settings ) as $setting ) {
+			$default_name = "default_$setting"; // e.g. default_dequeue_max_bytes
+			$settings[ $setting ] = (int) get_option( self::SETTINGS_OPTION_PREFIX.$setting, Jetpack_Sync_Defaults::$$default_name );
+		}
+		return $settings;
+	}
+
+	function update_settings( $new_settings ) {
+		$validated_settings = array_intersect_key( $new_settings, self::$valid_settings );
+		foreach( $validated_settings as $setting => $value ) {
+			update_option( self::SETTINGS_OPTION_PREFIX.$setting, $value, true );
+		}
+	}
+
+	function update_options_whitelist() {
+		$this->options_whitelist = apply_filters( 'jetpack_options_whitelist', Jetpack_Sync_Defaults::$default_options_whitelist );
```

Comment:
This needs a docblock.

`/** This filter is already documented in json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php */`